custom/icds_reports/tests/test_aggregation_script.py

```diff
@@ -0,0 +1,212 @@
+                >>> self._convert_decimal_to_string(Decimal('0.0000000'))
+                '0.0000000'
+        """
+        value_str = str(value)
+        p = re.compile('0E-(?P<zeros>[0-9]+)')
+        match = p.match(value_str)
+        if match:
+            return '0.{}'.format(int(match.group('zeros')) * '0')
+        else:
+            return value_str
+
+    def _load_data_from_db(self, table_name, sort_key, filter_by=None):
+        session_helper = connection_manager.get_session_helper(ICDS_UCR_CITUS_ENGINE_ID)
+        engine = session_helper.engine
+        session = session_helper.Session
+        metadata = sqlalchemy.MetaData(bind=engine)
+        metadata.reflect(bind=engine)
+        table = metadata.tables[table_name]
+        columns = [
+            column.name
+            for column in table.columns
+        ]
+        with engine.begin() as connection:
+            query = session.query(table).order_by(*sort_key)
+            if filter_by:
+                query = query.filter_by(**filter_by)
+            rows = query.all()
+            for row in rows:
+                row = list(row)
+                for idx, value in enumerate(row):
+                    if isinstance(value, date):
+                        row[idx] = value.strftime('%Y-%m-%d')
+                    elif isinstance(value, time):
+                        row[idx] = value.strftime("%H:%M:%S.%f").rstrip('0').rstrip('.')
+                    elif isinstance(value, int):
+                        row[idx] = str(value)
+                    elif isinstance(value, (float, Decimal)):
+                        row[idx] = self._convert_decimal_to_string(row[idx])
+                    elif value is None:
+                        row[idx] = ''
+                yield dict(zip(columns, row))
+
+    def _load_and_compare_data(self, table_name, path, sort_key=None, filter_by=None):
+        # To speed up tests, we use a sort_key wherever possible
+        #   to presort before comparing data
+        if sort_key:
+            self._fasterAssertListEqual(
+                list(self._load_data_from_db(table_name, sort_key, filter_by)),
+                self._load_csv(path)
+            )
+        else:
+            sort_key = lambda x: x
+            self._fasterAssertListEqual(
+                sorted(
+                    list(self._load_data_from_db(table_name, [], filter_by)),
+                    key=sort_key
+                ),
+                sorted(
+                    self._load_csv(path),
+                    key=sort_key
+                )
+            )
+
+
+class CcsRecordMonthlyAggregationTest(AggregationScriptTestBase):
+    always_include_columns = {'awc_id', 'case_id'}
+    fixtures_needed = [
+        'ccs_cases.csv',
+        'pregnant_tasks.csv',
+        'person_cases.csv',
+        'thr_form.csv',
+        'birth_preparedness.csv',
+        'pnc_forms.csv',
+        'complementary_feeding.csv',
+        'delivery_form.csv',
+    ]
```

Comment:
These are all of the required fixtures for this task. 

In the future, I wouldn't want this to just be limited to UCRs. For example when `agg_child_heatlh` is tested like this, it would make sense to have `child_health_monthly` as a "fixture" instead of a list of UCRs.